### DISCUSSION TITLE
What is the type of session?

### DISCUSSION THREAD

**Discussion by Karthik777** [AnswerDotAI/fasthtml#390] [Q&A]
Please let me know if this is not a FastHTML question.

I'm writing a fasthtml app to showcase it's potential to my company execs. python type hinting is a must for us. I've been trying to understand the type to use for session. I can see it's a dict. But, if I type hint it. it doesn't work. it's either empty (inside before for the first call) or returns a 400.
I've tried tuple, Mapping and MutableMapping.

This works:
```
def before(session):
    if 'sid' not in session:
        session['sid'] = str(uuid.uuid4())
```

This does not: I get a 400.

```
def before(session: tuple | dict):
    if 'sid' not in session:
        session['sid'] = str(uuid.uuid4())
```

some screenshots:
without type hinting
![image](https://github.com/user-attachments/assets/98e51100-292d-45d5-b988-48429f3c3792)
with type hinting
![image](https://github.com/user-attachments/assets/0875d1fb-e91f-4313-ae37-b9d119d26c20)

**Comment by Jamsjz** [comment]
The type for session is good old dictionary.
You can find the docs here: https://docs.fastht.ml/tutorials/quickstart_for_web_devs.html#sessions
